From 2595b940811bd401b5f6438a277363a4cb0d56c8 Mon Sep 17 00:00:00 2001
From: Tom Augspurger <tom.w.augspurger@gmail.com>
Date: Fri, 17 Jul 2020 16:02:40 -0500
Subject: [PATCH] PKG: No error for osx

---
 setup.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index aebbdbf4d1..c2417ccae7 100755
--- a/setup.py
+++ b/setup.py
@@ -433,7 +433,9 @@ if is_platform_windows():
         extra_compile_args.append("/Z7")
         extra_link_args.append("/DEBUG")
 else:
-    extra_compile_args = ["-Werror"]
+    extra_compile_args = []
+    if not is_platform_mac():
+        extra_compile_args.append("-Werror")
     extra_link_args = []
     if debugging_symbols_requested:
         extra_compile_args.append("-g")
-- 
2.23.0

